src/components/common/markets_dropdown.tsx

```diff
@@ -193,129 +211,84 @@ const FILTER_TOKENS = ['All', 'ETH', 'DAI', 'USDC'];
+        symbol: 'dgd',
         price: '0.643',
         previousDay: '78',
         currentDay: '90',
         dayVol: '456.23',
     },
     {
         name: 'ZRX / DAI',
+        symbol: 'mln',
         price: '0.978687',
         previousDay: '12',
         currentDay: '26',
         dayVol: '24534.56',
     },
-    {
-        name: 'ABC / ETH',
-        price: '0.755',
-        previousDay: '78',
-        currentDay: '78',
-        dayVol: '515235.05',
-    },
-    {
-        name: 'CDF / ETH',
-        price: '0.7547',
-        previousDay: '56',
-        currentDay: '78',
-        dayVol: '515235.00',
-    },
-    {
-        name: 'GHI / ETH',
-        price: '0.765',
-        previousDay: '90',
-        currentDay: '80',
-        dayVol: '3948573.34',
-    },
-    {
-        name: 'OMG / ETH',
-        price: '0.908',
-        previousDay: '55',
-        currentDay: '55',
-        dayVol: '3455.00',
-    },
-    {
-        name: 'OMG / DAI',
-        price: '0.765',
-        previousDay: '99',
-        currentDay: '101',
-        dayVol: '123235.00',
-    },
-    {
-        name: 'ASD / ETH',
-        price: '0.543',
-        previousDay: '90',
-        currentDay: '88',
-        dayVol: '1515235.00',
-    },
-    {
-        name: 'ZRX / DAI',
-        price: '0.978687',
-        previousDay: '12',
-        currentDay: '34',
-        dayVol: '24534.56',
-    },
-    {
-        name: 'ABC / ETH',
-        price: '0.755',
-        previousDay: '78',
-        currentDay: '98',
-        dayVol: '515235.05',
-    },
-    {
-        name: 'CDF / ETH',
-        price: '0.7547',
-        previousDay: '56',
-        currentDay: '78',
-        dayVol: '515235.00',
-    },
-    {
-        name: 'GHI / ETH',
-        price: '0.765',
-        previousDay: '90',
-        currentDay: '80',
-        dayVol: '3948573.34',
-    },
 ];
 
-export class MarketsDropdown extends React.Component<Props, State> {
+class MarketsDropdown extends React.Component<Props, State> {
     public readonly state: State = {
         // Note: this will give you a headache in the long run, so please use redux / mobx or something...
         isLoadingMarkets: true,
         selectedFilter: 0,
-        selectedMarketItem: 0,
+        filteredTokens: [],
+        alreadyAssignedProps: false,
+        search: '',
     };
 
     private _closeDropdown: any;
 
+    public static getDerivedStateFromProps(nextProps: Props, previousState: State): any {
+        // Check for initialization
+        if (!previousState.alreadyAssignedProps && nextProps.tokens && nextProps.tokens.length > 0) {
+            previousState.filteredTokens = nextProps.tokens;
+            previousState.alreadyAssignedProps = true;
+            return previousState;
+        }
+        return null;
+    }
+
     public render = () => {
-        const { ...restProps } = this.props;
+        const { tokens, selectedToken, ...restProps } = this.props;
+
+        const token = tokens.find(obj => {
```

Comment:
I think you don't need to do the `find` on `tokens`. With:
```typescript
const tokenName = (selectedToken && selectedToken.name) || '';
```
should be enough.

Reply:
@unjapones Done! Thanks